pkg/controller/sparkapplication/controller.go

```diff
@@ -643,7 +644,7 @@ func (c *Controller) submitSparkApplication(app *v1beta1.SparkApplication) *v1be
 		DriverInfo: v1beta1.DriverInfo{
 			PodName: driverPodName,
 		},
-		SubmissionAttempts:        app.Status.SubmissionAttempts + 1,
+		SubmissionAttempts:        0, // Reset in case of a successfuly submission.
```

Comment:
Given that we are now resetting in `PendingRerun` we don't need this anymore. Can we please revert this ?

This way, SubmissionAttempts will still reflect correct attempt count for a single run of the application without impacting any retry behavior.

Reply:
Just like we reset `ExecutionAttempts` upon a successful run, I think the right semantic is to reset `SubmissionAttempts` upon a successful submission as `SubmissionAttempts` is really the number of consecutive failed submission attempts.